nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/PipelineXml.java

```diff
@@ -0,0 +1,451 @@
+
+            PipelineXmlData pd = null;
+            try {
+                if (subject == XML_PIPELINE_CONFIG.getName()) {
+                    pd = new PipelineXmlData(input);
+                } else if (subject == XML_PIPELINE_FILE.getName()) {
+                    pd = new PipelineXmlData(new Input(input));
+                } else {
+                    return builder.valid(false).explanation("Can only validate XML Pipeline Config or XML Pipeline File").build();
+                }
+            } catch (SaxonApiException | XProcException e) {
+                return builder.valid(false).explanation("XProc pipeline is invalid: " + e).build();
+            }
+
+            final String inputPort = findPrimaryInputPort(pd.pipeline);
+            if (inputPort == null) {
+                return builder.valid(false).explanation("XProc pipeline must define a primary non-parameter input port").build();
+            }
+
+            return builder.valid(true).build();
+        }
+    };
+
+    private static final Validator XPROC_OPTION_NAME_VALIDATOR = new Validator() {
+        @Override
+        public ValidationResult validate(String optName, String optValue, ValidationContext context) {
+            final ValidationResult.Builder builder = new ValidationResult.Builder();
+            builder.subject(optName).input(optValue);
+
+            try {
+                QName.fromClarkName(optName);
+                builder.valid(true);
+            } catch (IllegalArgumentException e) {
+                builder.valid(false).explanation("Option name must be of the form {uri}local.");
+            }
+
+            return builder.build();
+        }
+    };
+
+    // note that the names of these properties and relationships intentionally
+    // have spaces so they cannot conflict with XProc option and output port
+    // names, which are restricted to QName's and NCName's
+    public static final PropertyDescriptor XML_PIPELINE_FILE = new PropertyDescriptor.Builder()
+            .name("xml pipeline file")
+            .displayName("XML Pipeline File")
+            .description("Full path to a file containing the XProc XML pipeline configuration. Only one of XML Pipeline File or XML Pipeline Config may be used.")
+            .required(false)
+            .addValidator(StandardValidators.FILE_EXISTS_VALIDATOR)
+            .addValidator(XML_PIPELINE_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor XML_PIPELINE_CONFIG = new PropertyDescriptor.Builder()
+            .name("xml pipeline config")
+            .displayName("XML Pipeline Config")
+            .description("XProc XML pipeline configuration. Only one of XML Pipeline File or XML Pipeline Config may be used.")
+            .required(false)
+            .addValidator(Validator.VALID)
+            .addValidator(XML_PIPELINE_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor XML_PIPELINE_POOL_SIZE = new PropertyDescriptor.Builder()
```

Comment:
This pool size configuration option is really not necessary. Any size over the number of concurrent tasks will lead to pool instances never being used and just wasting memory. And any size under the number of concurrent tasks will effectively limit how many concurrent tasks can run at once. So instead of making this configurable, the pool size should just be initialized to the maximum number of concurrent tasks.